client/client/src/EntryFormView.tsx

```diff
@@ -0,0 +1,122 @@
+import { FormEvent } from 'react';
+import { useState } from 'react';
+import './data';
+import { Entry, addEntry, removeEntry, updateEntry } from './data';
+import DeleteEntryModal from './DeleteEntryModal';
+type Props = {
+  targetEntry: Entry | null;
+  handleViewSwap: () => void;
+};
+export default function EntryFormView({ targetEntry, handleViewSwap }: Props) {
+  const [formTitle, setFormTitle] = useState(targetEntry?.formTitle ?? '');
+  const [formURL, setFormURL] = useState(targetEntry?.formURL ?? '');
+  const [formNotes, setFormNotes] = useState(targetEntry?.formNotes ?? '');
+  const [modalView, setModalView] = useState(false);
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+    const inputtedEntries = Object.fromEntries(formData.entries());
+    if (targetEntry) {
+      inputtedEntries.entryId = targetEntry.entryId;
+      updateEntry(inputtedEntries);
+    } else {
+      addEntry(inputtedEntries);
+    }
+    event.currentTarget.reset();
+    {
+      handleViewSwap();
```

Comment:
re: [my comment ](https://github.com/aehrenkranz/code-journal-react/pull/5/files#r1458219705)in App.tsx:

we can use `setEditing(undefined); ` instead!